ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py

```diff
@@ -76,3 +76,85 @@ def test_binary_cross_entropy_with_logits(
         pos_weight=pos_weight[0],
         reduction=reduction,
     )
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.ctc_loss",
+    dtype_and_logits=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_value=-10,
+        max_value=10,
+        allow_inf=False,
+        exclude_min=True,
+        exclude_max=True,
+        min_num_dims=3,
+        max_num_dims=3,
+        min_dim_size=2,
+        ),
+
+    dtype_and_targets=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        min_value=0,
+        max_value=9,
+        allow_inf=False,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=2,
+    ),
+
+    logit_lengths=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        min_value=1,
+        max_value=50,
+        allow_inf=False,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=2,
+    ),
+    label_lengths=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+        min_value=1,
+        max_value=50,
+        allow_inf=False,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=2,
+    ),
+    blank=st.integers(min_value=0, max_value=9),
+    reduction=st.sampled_from(["none", "sum", "mean"]),
+    zero_infinity=st.booleans(),
+)
+def test_ctc_loss(
+    dtype_and_logits,
+    dtype_and_targets,
+    logit_lengths,
+    label_lengths,
+    blank,
+    reduction,
+    zero_infinity,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    #logits_dtype, logits = dtype_and_logits
+    #targets_dtype, targets = dtype_and_targets
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=dtype_and_logits[0] + dtype_and_targets[0] + logit_lengths[0] + label_lengths[0] + [np.int32],
```

Comment:
pass the `np.int32` as a string instead
```suggestion
        input_dtypes=dtype_and_logits[0] + dtype_and_targets[0] + logit_lengths[0] + label_lengths[0] +"int32",
```

Reply:
but then I get the error: TypeError: can only concatenate list (not "str") to list

Reply:
I think it is because of the helpers.dtype_and_values which returns:
A strategy that draws a tuple of a list of dtypes and a list of their respective arrays.

Reply:
maybe [np.int32] was right at the first place, what do you think?

Reply:
and is there a way to push the changes without this command:
git reset --hard upstream/master
because of this I add the codes manually all over again every time I make a commit:)